src/Core/Utils/Attribs.hpp

```diff
@@ -193,54 +209,35 @@ class AttribHandle
  * \warning There is no error check on the handles attribute type.
  *
  */
-class RA_CORE_API AttribManager
+class RA_CORE_API AttribManager : public Observable<const std::string&>
 {
   public:
     using value_type         = AttribBase;
     using pointer_type       = value_type*;
     using smart_pointer_type = std::unique_ptr<value_type>;
     using Container          = std::vector<smart_pointer_type>;
 
-    AttribManager() {}
+    inline AttribManager();
 
     /// Copy constructor and assignment operator are forbidden.
     AttribManager( const AttribManager& m ) = delete;
     AttribManager& operator=( const AttribManager& m ) = delete;
 
-    AttribManager( AttribManager&& m ) :
-        m_attribs( std::move( m.m_attribs ) ),
-        m_attribsIndex( std::move( m.m_attribsIndex ) ) {}
-
-    AttribManager& operator=( AttribManager&& m ) {
-        m_attribs      = std::move( m.m_attribs );
-        m_attribsIndex = std::move( m.m_attribsIndex );
-        return *this;
-    }
+    inline AttribManager( AttribManager&& m );
 
-    ~AttribManager() { clear(); }
+    inline AttribManager& operator=( AttribManager&& m );
+    ~AttribManager();
```

Comment:
Why would you want it virtual since there is no other virtual method in this class?

Reply:
Because `AttribManager` inherits another class.
And when you use inheritance you want to use virtual destructors to avoid memory leaks.
https://www.quantstart.com/articles/C-Virtual-Destructors-How-to-Avoid-Memory-Leaks/

Reply:
you don't have to declare it virtual. If the base class has a virtual destructor, derived class destructor's are implicetly virtual. Nevertheless, it is a good practive to precise override in the destructor.

Reply:
Very interesting discussion, but it's all a question of point of view.

My proposition is to follow clangtidy recommandation (which I don't know what it is).

or https://en.cppreference.com/w/cpp/language/destructor which do not mention override

Reply:
I checked again, and yes, from aug 1st, the recommendation is to user override.
Until the next change.

So *my* point of view is to have either override or virtual (or final), but not both, and not any ... to clearly state the dtor is virtual.

Reply:
I do agree that `override` is better than `virtual` for the inheriting class, but the base class must declare its destructor as `virtual`.
What I like with `override` is that clearly says that the destruction will be propagated up to the base class.

Btw, the class Observable has no virtual destructor, while it is a base class inherited multiple times.

Reply:
the ref for the claim of @dlyr : 
http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-override

It is recommended to have override, virtual keyword must be reserved to introduce a new virtual fonction in the hierarchy. override clearly state what is the intent of the method that remains virtual. And if the method is not virtual, an error is generated at compile.

From the C++ Standard (10.3 Virtual functions)

> 6 Even though destructors are not inherited, a destructor in a derived class overrides a base class destructor declared virtual; see 12.4 and 12.5.

Reply:
```suggestion
    ~AttribManager() override;
```